src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/source_filters_table.js

```diff
@@ -1,128 +1,185 @@
+        const filterFilterToLowercase = filterFilter.toLowerCase();
+        return filters.filter(filter =>
+          filter.value.toLowerCase().includes(filterFilterToLowercase)
+        );
+      }
+
+      return filters;
+    }
+  );
+
+  startDeleteFilter = filter => {
+    this.setState({
+      filterToDelete: filter,
+      isDeleteConfirmationModalVisible: true,
+    });
+  };
+
+  hideDeleteConfirmationModal = () => {
+    this.setState({
+      filterToDelete: undefined,
+      isDeleteConfirmationModalVisible: false,
+    });
+  };
+
+  deleteFilter = async () => {
+    const { indexPattern, onAddOrRemoveFilter } = this.props;
+    const { filterToDelete, filters } = this.state;
+
+    indexPattern.sourceFilters = filters.filter(filter => {
+      return filter.clientId !== filterToDelete.clientId;
+    });
+
+    this.setState({ isSaving: true });
+    await indexPattern.save();
+    onAddOrRemoveFilter && onAddOrRemoveFilter();
+    this.updateFilters();
+    this.setState({ isSaving: false });
+    this.hideDeleteConfirmationModal();
+  };
+
+  onAddFilter = async value => {
+    const { indexPattern, onAddOrRemoveFilter } = this.props;
+
+    indexPattern.sourceFilters = [
+      ...(indexPattern.sourceFilters || []),
+      { value },
+    ];
+
+    this.setState({ isSaving: true });
+    await indexPattern.save();
+    onAddOrRemoveFilter && onAddOrRemoveFilter();
+    this.updateFilters();
+    this.setState({ isSaving: false });
+  };
+
+  saveFilter = async ({ filterId, newFilterValue }) => {
+    const { indexPattern } = this.props;
+    const { filters } = this.state;
+
+    indexPattern.sourceFilters = filters.map(filter => {
+      if (filter.clientId === filterId) {
+        return {
+          value: newFilterValue,
+          clientId: filter.clientId,
+        };
+      }
+      return filter;
+    });
+
+    this.setState({ isSaving: true });
+    await indexPattern.save();
+    this.updateFilters();
+    this.setState({ isSaving: false });
+  };
+
+  renderDeleteConfirmationModal() {
+    const { filterToDelete } = this.state;
+
+    if (!filterToDelete) {
+      return null;
+    }
+
+    return (
+      <EuiOverlayMask>
+        <EuiConfirmModal
+          title={`Delete source filter '${filterToDelete.value}'?`}
+          onCancel={this.hideDeleteConfirmationModal}
+          onConfirm={this.deleteFilter}
+          cancelButtonText="Cancel"
+          confirmButtonText="Delete"
+          defaultFocusedButton={EUI_MODAL_CONFIRM_BUTTON}
+        />
+      </EuiOverlayMask>
+    );
+  }
+
+  render() {
+    const { indexPattern, fieldWildcardMatcher } = this.props;
+
+    const { isSaving } = this.state;
+
+    const filteredFilters = this.getFilteredFilters(this.state, this.props);
+
+    return (
+      <div>
+        <Header />
+        <AddFilter onAddFilter={this.onAddFilter} />
+        <EuiSpacer size="l" />
+        {isSaving ? <EuiLoadingSpinner /> : null}
+        {filteredFilters.length > 0 ? (
```

Comment:
And can we always show the table? It should display a "no items" message if it's empty, I think.